crates/lib3h/src/engine/real_engine.rs

```diff
@@ -155,20 +193,91 @@ impl<'engine> RealEngine<'engine> {
         }
         Ok(())
     }
+
+    ///
+    pub fn get_this_peer_sync(&mut self, maybe_chainId: Option<ChainId>) -> PeerData {
+        trace!("engine.get_this_peer_sync() ...");
+        let gateway = if let Some(chain_id) = maybe_chainId {
+            self.space_gateway_map
+                .get_mut(&chain_id)
+                .expect("Should have the space gateway")
+        } else {
+            &mut self.network_gateway
+        };
+        gateway
+            .request(
+                Lib3hTrace,
+                GatewayRequestToChild::Dht(DhtRequestToChild::RequestThisPeer),
```

Comment:
The result of this call is precisely what we put in when creating this gateway, no?

If we want to keep our actor model pure and not make concrete state calls on the gateway actors directly, I'd say we should wrap the references with an internal struct here that can keep additional information like this so we don't have to make an async request to get it.

Alternately, if we want to side-step the actor model a bit in the short-term, I'd say making a direct request on the concrete gateway would be better than making a manual process call here in this helper function.

Reply:
not allways. In one occasion I need the up-to-date timestamp to know if I need to broadcast myself